logger/mask.go

```diff
@@ -189,6 +190,10 @@ func getMaskedValue(maskChar, valueToReplace string, toMask MaskData) string {
 		return valueToReplace
 	}
 
+	if toMask.UseMaskChar {
+		return valueToReplace[:toMask.FirstCharsToShow] + maskChar + valueToReplace[lastIndexToShowStart:]
+	}
```

Comment:
this is similar to line 209 right?

Reply:
Similar, yes. But not the same intention. `UseMaskChar` will just blindly use the `maskChar` passed in. Like a pure replacement.

In the case of API keys, being able to count the number of masked characters can still allow for guesswork.

Using `maskChar`, caller to the `getMaskedValue` function will just get the replacement immediately, so that it makes it difficult to guess.

`my_ap******123` vs `my_ap***123` where the former is the full API key and the latter is a better masked one that prevents guess work.

Reply:
oh I see because your `maskChar` is `***`

Reply:
Perhaps we can have a better name for the `UseMaskChar` for better readability and understanding.

Reply:
How about `KeepSameLength`?
- `true`: normal masking
- `false`: current `UseMaskChar = true` logic

Reply:
That would mean that for other use cases we'll need to set `KeepSameLength: true` since zero value would be `false` by default.

Personally, I think it is OK to go with this approach, and it allows the expansion at times where we don't want to keep the same length like in this case. What do you think?